src/service/file_manager/file.factory.ts

```diff
@@ -15,7 +15,9 @@ export class FileFactory {
     const organizationId = fileProperties.organizationId.toString();
     const userId = fileProperties.userId.toString();
 
-    const key = path.resolve('organization', organizationId, 'user', userId, folder, filename);
+    const locationElements = ['organization', organizationId, 'user', userId, folder];
+    if (filename) locationElements.push(filename);
+    const key = path.posix.join(...locationElements);
```

Comment:
```suggestion
  const key = path.resolve('organization', organizationId, 'user', userId, folder, filename);
```

Reply:
```suggestion
const key = path.posix.join('organization', organizationId, 'user', userId, folder, filename);
```
"resolve" does output system's absolute path. We just want to join it